tests/saliency_metrics/metrics/insertion_deletion/test_insertion_deletion.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+import torch
+import torch.nn as nn
+
+from saliency_metrics.metrics.insertion_deletion.insertion_deletion import InsertionDeletion
+from saliency_metrics.models import CUSTOM_CLASSIFIERS
+
+
+@CUSTOM_CLASSIFIERS.register_module()
+class TestNet(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.features = nn.Sequential(
+            nn.Conv2d(in_channels=3, out_channels=5, kernel_size=3, padding=1),
+            nn.MaxPool2d(2, 2),
+            nn.ReLU(inplace=True),
+        )
+        self.classifier = nn.Sequential(nn.Linear(16 * 16 * 5, 10))
+        self.initialize_weights()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.features(x)
+        x = x.view(-1, 16 * 16 * 5)
+        x = self.classifier(x)
+        return x
+
+    def initialize_weights(self) -> None:
+        for i, x in enumerate(self.parameters()):
+            prng = np.random.RandomState(i)
+            new_parameters = prng.uniform(0, 0.02, x.shape)
+            x.data = torch.tensor(new_parameters, dtype=torch.float32)
+
+
+@pytest.fixture
+def dummy_img_and_smap():
+    # three-channel image
+    img = torch.linspace(0, 1, 3072, dtype=torch.float32).reshape(1, 3, 32, 32)
+    # saliency map
+    smap = torch.linspace(0, 1, 1024, dtype=torch.float32).reshape(32, 32)
+    yield img, smap
+
+
+@pytest.mark.parametrize("perturb_step_size", [0, 10])
+@pytest.mark.parametrize("forward_batch_size", [0, 32])
+def test_init(perturb_step_size, forward_batch_size):
+    classifier_cfg = dict(type="custom.TestNet")
+    if forward_batch_size == 0 or perturb_step_size == 0:
+        with pytest.raises(ValueError):
+            ins_del = InsertionDeletion(
+                classifier_cfg, forward_batch_size=forward_batch_size, perturb_step_size=perturb_step_size
+            )
+    else:
+        ins_del = InsertionDeletion(
+            classifier_cfg, forward_batch_size=forward_batch_size, perturb_step_size=perturb_step_size
+        )
+        assert ins_del.forward_batch_size == forward_batch_size
+        assert ins_del.perturb_step_size == perturb_step_size
+
```

Comment:
Please check 
1. If the instance has a gaussian blur object. 
2. If the classifier is under eval mode and all the params are frozen.